pkg/collector/corechecks/snmp/checkconfig/config_utils.go

```diff
@@ -0,0 +1,26 @@
+package checkconfig
+
+import (
+	"hash/fnv"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+)
+
+func buildDeviceID(origTags []string) (string, []string) {
+	h := fnv.New64()
+	var tags []string
+	for _, tag := range origTags {
```

Comment:
Add a comment here https://github.com/DataDog/datadog-agent/pull/8971/commits/7d81e83f0217ed39b22b4753cc574c5befd60ffc

```

		// When using snmp_listener, the `autodiscovery_subnet` tag is passed
		// via autodiscovery template. We are ignoring it to avoid deviceID
		// being changed when we change subnet e.g. 10.10.0.0/29 -> 10.10.0.0/30
```